### PR TITLE
[Feat] Rework all Datasource apis + a lot of tests

### PR DESCRIPTION
**Pull request by smnandre**
Sorry for the size, I will split after discussion

### CODE REVIEW

File: src/Domain/Value/DatasourceEntry.php [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+            TrimmedNonEmptyString::fromString($dimensionValue);
+        }
+    }
+
+    /**
+     * @param array<string, mixed> $values
+     */
+    public static function fromArray(array $values): self
+    {
+        Assert::same(['id', 'name', 'value', 'dimension_value'], array_keys($values));
```
Comment (OskarStark) [inline src/Domain/Value/DatasourceEntry.php]: 👍🏻

File: src/DatasourceEntriesApi.php [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        Assert::regex($datasource, '/^[a-z0-9-]+$/');
+        Assert::regex($dimension, '/^[a-z0-9-]+$/');
+
+        return $this->getDatasourceEntries(['datasource' => $datasource, 'dimension' => $dimension], $pagination);
+    }
+
+    /**
+     * @param array<string, mixed> $parameters
+     */
+    private function getDatasourceEntries(array $parameters, ?Pagination $pagination = null): DatasourceEntriesResponse
```
Comment (silasjoisten) [inline src/DatasourceEntriesApi.php]: `collectionRequest()` ?

File: src/DatasourceEntriesApi.php [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    }
+
+    public function allByDimension(string $dimension, ?Pagination $pagination = null): DatasourceEntriesResponse
+    {
+        Assert::regex($dimension, '/^[a-z0-9-]+$/');
+
+        return $this->getDatasourceEntries(['datasource' => $dimension], $pagination);
+    }
+
+    public function allByDatasourceDimension(string $datasource, string $dimension, ?Pagination $pagination = null): DatasourceEntriesResponse
```
Comment (silasjoisten) [inline src/DatasourceEntriesApi.php:50]: is this needed at all? maybe we shouldnt allow this here. i think this belongs more to the Datasource Api not the entries.

File: src/Domain/Value/DatasourceEntry.php (hunk 1 above)
Comment (silasjoisten) [reply src/Domain/Value/DatasourceEntry.php]: you only assert that the keys exist but whats with its value like is it trimmed? is it non empty string? does it needs a certain min length etc. this value object is not strict enough

Comment (smnandre) [reply src/Domain/Value/DatasourceEntry.php]: Done in the constructor :)

File: src/DatasourceEntriesApi.php (hunk 3 above)
Comment (smnandre) [reply src/DatasourceEntriesApi.php:50]: It returns a list of entries, i feel it's very similar to byContentType for instance in the StoriesApi

**Pull request merged**